src/core/CMakeLists.txt

```diff
@@ -2196,6 +2196,11 @@ else()
   endif()
 endif()
 
+# Workaround CLANG deprecation warning
+if ("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
+  set_source_files_properties(painting/qgsmaskpaintdevice.cpp PROPERTIES COMPILE_FLAGS "-Wno-deprecated-declarations")
```

Comment:
strange... What are those warnings? Can we not fix them?

Reply:
feel free to step in... https://cdash.orfeo-toolbox.org/viewBuildError.php?type=1&buildid=27379
I didn't succeed with the `Q_NOWARN_DEPRECATED_PUSH` macro, but your experience may help :+1:

Reply:
try to surround reference to QgsMaskPaintEngine with Q_NOWARN_DEPRECATED_PUSH and Q_NOWARN_DEPRECATED_POP, starting maybe with the declaration on line 89 of qgsmaskpaintdevice.h

Maybe we also should deprecate the whole classes Engine & Device.

Reply:
That's what I already tried - it didn't change neither to include the whole two deprecated classes, neither even in combination with Q_NOWARN_DEPRECATED_PUSH/POPping the calls in the qgsmaskpaintdevice.cpp :-( - that's why I turned to this solution which works after all...

Reply:
tested it myself and fail to find a way to make it work. OK then